src/algebra/homology/additive.lean

```diff
@@ -0,0 +1,168 @@
+
+instance homology_additive : (homology_functor V c i).additive :=
+{ map_zero' := λ C D, begin
+    dsimp [homology_functor],
+    ext,
+    simp only [limits.cokernel.π_desc, limits.comp_zero, homology.π_map],
+    convert zero_comp,
+    ext,
+    simp,
+  end,
+  map_add' := λ C D f g, begin
+    dsimp [homology_functor],
+    ext,
+    simp only [homology.π_map, preadditive.comp_add, ←preadditive.add_comp],
+    congr,
+    ext, simp,
+  end }
+
+end homological_complex
+
+namespace category_theory
+
+variables {W : Type*} [category W] [preadditive W]
+
+/--
+An additive functor induces a functor between homological complexes.
+This is sometimes called the "prolongation".
+-/
+@[simps]
+def functor.map_homological_complex (F : V ⥤ W) [F.additive] (c : complex_shape ι) :
+  homological_complex V c ⥤ homological_complex W c :=
+{ obj := λ C,
+  { X := λ i, F.obj (C.X i),
+    d := λ i j, F.map (C.d i j),
+    shape' := λ i j w, by rw [C.shape _ _ w, F.map_zero],
+    d_comp_d' := λ i j k, by rw [←F.map_comp, C.d_comp_d, F.map_zero], },
+  map := λ C D f,
+  { f := λ i, F.map (f.f i),
+    comm' := λ i j, by { dsimp,  rw [←F.map_comp, ←F.map_comp, f.comm], }, }, }.
+
+instance functor.map_homogical_complex_additive
+  (F : V ⥤ W) [F.additive] (c : complex_shape ι) : (F.map_homological_complex c).additive := {}
+
+/--
+A natural transformation between functors induces a natural transformation
+between those functors applied to homological complexes.
+-/
+@[simps]
+def nat_trans.map_homological_complex {F G : V ⥤ W} [F.additive] [G.additive]
+  (α : F ⟶ G) (c : complex_shape ι) : F.map_homological_complex c ⟶ G.map_homological_complex c :=
+{ app := λ C, { f := λ i, α.app _, }, }
+
+@[simp] lemma nat_trans.map_homological_complex_id (c : complex_shape ι) (F : V ⥤ W) [F.additive] :
+  nat_trans.map_homological_complex (𝟙 F) c = 𝟙 (F.map_homological_complex c) :=
+by tidy
+
+@[simp] lemma nat_trans.map_homological_complex_comp (c : complex_shape ι)
+  {F G H : V ⥤ W} [F.additive] [G.additive] [H.additive]
+  (α : F ⟶ G) (β : G ⟶ H):
+  nat_trans.map_homological_complex (α ≫ β) c =
+    nat_trans.map_homological_complex α c ≫ nat_trans.map_homological_complex β c :=
+by tidy
+
+@[simp, reassoc] lemma nat_trans.map_homological_complex_naturality {c : complex_shape ι}
+  {F G : V ⥤ W} [F.additive] [G.additive] (α : F ⟶ G) {C D : homological_complex V c} (f : C ⟶ D) :
+  (F.map_homological_complex c).map f ≫ (nat_trans.map_homological_complex α c).app D =
+    (nat_trans.map_homological_complex α c).app C ≫ (G.map_homological_complex c).map f :=
+by tidy
+
+end category_theory
+
+namespace chain_complex
+
+variables [has_zero_object V] {W : Type*} [category W] [preadditive W] [has_zero_object W]
+
+/--
+Turning an object into a complex supported at zero than applying a functor is
+the same as applying the functor then forming the complex.
+-/
+def single₀_map_homological_complex (F : V ⥤ W) [F.additive] :
```

Comment:
`@[simps]`?

Reply:
It turns out better to construct explicit simp lemmas, to do the zero/succ case distinction.